Ints view

JackLeeHal opened the pull request:
Fixes # .

Changes proposed in this pull request:
- Generate Integration view
- Use antlr to parse epfmt and appfmt

anzdaddy committed:
Configure golangci-lint & address its warnings

Commit:
Generate integration diagram

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+}
+
+type DependencySet struct {
+	Slice  []*AppDependency
```
Why not use map to backend the Set?

JackLeeHal replied:
fixed

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	}
+}
+
+func (set *DependencySet) Contains(dep *AppDependency) bool {
```
With map backend, no need to search linearly.

JackLeeHal replied:
fixed

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	return cs.slice
+}
+
+func (cs *CallSlice) CollectCallStatement(stmts []*sysl.Statement) {
```
Prefer to CollectCallStatements

JackLeeHal replied:
fixed

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	r := []string{}
+	appReStr := toPattern(integrations.ToSlice())
+	re := regexp.MustCompile(appReStr)
+	for _, dep := range deps {
```
Why not use stream instead of for loop?

loki5265 replied:
import github.com/yale8848/stream to fix this.

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	return utils.MakeStrSet(r...)
+}
+
+func FindApps(module *sysl.Module, excludes, highlights utils.StrSet, deps []*AppDependency) utils.StrSet {
```
Looks quite similar with FindHighlights. May be refactored.

loki5265 replied:
fixed

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	targetName := target.Name
+	targetEndpoint := target.Endpoint
+	interApp := utils.MakeStrSet(targetName).Intersection(excludes)
+	interEndpoint := utils.MakeStrSet(targetEndpoint).Intersection(utils.MakeStrSet(".. * <- *", "*"))
```
does space impact? Is it true here that ".. * <- *" can be used as it is?

linwumeng replied:
Prefer to something like undeterminedEps or ignorableEps.

JackLeeHal replied:
fixed, space is sensitive, it should be ".. * <- *".

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	target := dep.Target
+	targetName := target.Name
+	targetEndpoint := target.Endpoint
+	interApp := utils.MakeStrSet(targetName).Intersection(excludes)
```
prefer to excludedApps.

JackLeeHal replied:
fixed

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	targetEndpoint := target.Endpoint
+	interApp := utils.MakeStrSet(targetName).Intersection(excludes)
+	interEndpoint := utils.MakeStrSet(targetEndpoint).Intersection(utils.MakeStrSet(".. * <- *", "*"))
+	// add to integration array
```
useless comment, or "Add to integration array since all dependencies are determined." would be better.

JackLeeHal replied:
fixed

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+			integrations.Add(dep)
+		}
+		// collect outbound dependencies
+		if lenPassthroughs > 0 &&
```
can we make comments here to explain cases?

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	return integrations
+}
+
+func (ds *DependencySet) CheckDependencies(module *sysl.Module) {
```
Prefer to ResolveDependencies.

JackLeeHal replied:
It was following python version's naming. Fixed

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	}
+}
+
+func (ds *DependencySet) handleStatement(stmts []*sysl.Statement, apps map[string]*sysl.Application, appname, epname string) {
```
Prefer to resolveStatementDependencies

JackLeeHal replied:
It was following python version's naming. Fixed

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+}
+
+func (dep *AppDependency) extractAppNames() utils.StrSet {
+	s := utils.StrSet{}
```
No sure the set implementation ensures order or not. Does order matter?

JackLeeHal replied:
order doesn't mater.

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+}
+
+func isSub(child, parent utils.StrSet) bool {
+	if parent == nil || len(parent) == 0 {
```
len(nil) == 0, so no need for "parent == nil"

linwumeng replied:
what if both parent and child are nil or empty?

loki5265 replied:
remove set == nil logic. add return true logic when both parent and child are nil or empty.

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	return len(child.Difference(parent)) == 0
+}
+
+func hasPattern(a string, arr []string) bool {
```
'a' is not a good name.

loki5265 replied:
rename it as 's'

linwumeng commented on sysl2/sysl/integration/deps.go:
```diff
@@ -0,0 +1,330 @@
+	return strings.Join(appname.Part, " :: ")
+}
+
+func arrayToMap(a []string) map[string]bool {
```
Prefer to buildStringBoolFilter

JackLeeHal replied:
fixed

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+
+type IntsDiagramVisitor struct {
+	m          *sysl.Module
+	sb         *strings.Builder
```
Why not io.Writer?

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+	stmtSlice []*sysl.Statement
+}
+
+type CallsDrawn struct {
```
Do you mean Edge?

JackLeeHal replied:
It was following python version's naming.

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+	return &Statements{[]*sysl.Statement{}}
+}
+
+func MakeIntsParam(apps []string, highlights map[string]struct{}, integrations []*AppDependency, app *sysl.Application, endpt *sysl.Endpoint) *IntsParam {
```
Prefer to dependencies instead of integrations.

JackLeeHal replied:
fixed

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+	attrs := getApplicationAttrs(visitor.m, appName)
+	attrs["appname"] = appName
+	label := utils.ParseFmt(attrs, visitor.m.Apps[visitor.project].GetAttrs()["appfmt"].GetS())
+	s := &_var{
```
useless local variable, s, here.

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+	var appName, alias, label string
+	attrs := map[string]string{}
+
+	appName = strings.Split(name, " : ")[0]
```
is it ensured the name containing ":"? is it possible to violate memory access?

JackLeeHal replied:
yes, it was construct this way before pass to this method

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+	return ts.topAlias
+}
+
+func (visitor *IntsDiagramVisitor) VarManagerForState(name string) string {
```
why not always use short name for receiver? Prefer to 'v'.

JackLeeHal replied:
fixed

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+	return ts.topAlias
+}
+
+func (visitor *IntsDiagramVisitor) VarManagerForState(name string) string {
```
Look like it is looking up alias or create it. Prefer to some name like 'LookupxxxxForzzzz' or 'LookupxxxByzzz'

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+	if s, ok := visitor.symbols[name]; ok {
+		return s.alias
+	}
+	i := len(visitor.symbols)
```
Look the default alias naming strategy is by indexer from 0.

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+	return s.alias
+}
+
+func (visitor *IntsDiagramVisitor) generateStateView(args *Args, viewParams viewParams, params *IntsParam) string {
```
Looks like binding values to template. So is it possible to use template instead of mixing statements and output together here. Like we use JSP to replace severlet.

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+		}
+	}
+	callsDrawn := map[CallsDrawn]struct{}{}
+	if viewParams.endptAttrs["view"].GetS() == "system" {
```
always lower case here?

JackLeeHal replied:
yes

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+		}
+
+	} else {
+		for _, v := range params.integrations {
```
try to remove duplicate code

linwumeng commented on sysl2/sysl/integration/viewGen.go:
```diff
@@ -0,0 +1,562 @@
+	}
+}
+
+func stringInSlice(a string, list []string) bool {
```
a,b are not good name. Maybe s for string, i for item would be better. stringInSlice is weird, maybe 'find' or 'lookup'.

linwumeng commented on sysl2/sysl/ints.go:
```diff
@@ -0,0 +1,115 @@
+	root_model, title, output, project, filter, modules string,
+	exclude []string, clustered, epa bool,
+) map[string]string {
+	r := make(map[string]string)
```
prefer to 'o' instead of 'r' which means outputs.

loki5265 replied:
I thought use 'r' as the shorthand for 'result'. It seems 'o' would be better than it, right?

linwumeng commented on sysl2/sysl/utils/strSet.go:
```diff
@@ -0,0 +1,129 @@
+func (s StrSet) ToSortedSlice() []string {
+	slice := s.ToSlice()
+	sorted := make([]string, len(slice))
+	copy(sorted, slice)
```
why make a copy?

loki5265 replied:
useless logic. remove the copy.

linwumeng commented on sysl2/sysl/utils/strSet.go:
```diff
@@ -0,0 +1,129 @@
+
+func (s StrSet) Difference(other StrSet) StrSet {
+	out := StrSet{}
+
```
if other is nil or empty, s can be returned if it is not nil or empty. think about these corner cases.

loki5265 replied:
add logic to handle empty case when other is empty.

linwumeng commented on sysl2/sysl/utils/strSet_test.go:
```diff
@@ -0,0 +1,162 @@
+	assert.Equal(t, []string{"a", "b", "e"}, c.ToSortedSlice(), "Unexpect result")
+}
+
+func TestDifference(t *testing.T) {
```
All miss boundary testing, nil and empty, and negative test cases.

loki5265 committed:
Fix issue #232 and #233 (#234)

* Fix issue #232 and #233

* Add default value for SYSL_PLANTUML
* Add kingpin package to handle command-line args
* Align command-line args with python

MichaelDao committed:
New documentation on installing plantuml server for mac osx (#238)

* New documentation on installing plantuml server for mac osx

* add gradle to travis

anzdaddy committed:
Add linters: bodyclose, depguard, dupl

anzdaddy committed:
Add linter gocritic

anzdaddy committed:
Add linter gofmt

anzdaddy committed:
Add linter goimports

anzdaddy committed:
Add linter interfacer

anzdaddy committed:
Add linter lll

anzdaddy committed:
Add linters: maligned, misspell

anzdaddy committed:
Add linters: nakedret, prealloc

anzdaddy committed:
Add linter: scopelint

anzdaddy committed:
Add linter: stylecheck

anzdaddy committed:
Add linter: unconvert

anzdaddy committed:
Add linter: unparam

anzdaddy committed:
Add linter: gochecknoinits + de-noise logging

anzdaddy committed:
Add linter: goconst

anzdaddy committed:
Add linter: gocyclo

anzdaddy committed:
Add linter: golint

anzdaddy committed:
Add linter: gochecknoglobals

anzdaddy committed:
Add linter: gosec

anzdaddy committed:
Simplify linter config (enable-all)

anzdaddy committed:
Add golangci-lint to .travis.yml

anzdaddy committed:
Add golangci-lint to .travis.yml install section

anzdaddy committed:
Temporarily suppress unparam for two functions.

anzdaddy committed:
Anzdaddy/golangci lint (#242)

Enable all golangci-lint linters

Temporarily suppress unparam for two functions

anzdaddy committed:
Move to golangci.com for linting (#253)

* Pin .golangci.yaml to golangci-lint 1.17.1
* Remove golangci-lint from travis build (golangci.com now does it).

anzdaddy committed:
Replace PySysl validation with compare-to-golden (#252)

Also remove SYSL_PYTHON_BIN from CI build

Allenhaoli committed:
sysl file tab issue

Allenhaoli committed:
fix ci issue
